### ISSUE TITLE
[Build]: Cannot access the file because it is being used by another process

### ISSUE THREAD

**Issue by ghost** [openvinotoolkit/openvino#26844]
### OpenVINO Version

2024.4.0-16579-c3152d32c9c-releases/2024/4

### Operating System

Windows System

### Hardware Architecture

x86 (64 bits)

### Target Platform

Host Name:                 LAPTOP-DE17RPQT
OS Name:                   Microsoft Windows 10 Home
OS Version:                10.0.19045 N/A Build 19045
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          -
Registered Organization:   HP
Product ID:                00325-97258-54349-AAOEM
Original Install Date:     9/26/2024, 9:07:47 AM
System Boot Time:          9/28/2024, 7:37:14 PM
System Manufacturer:       HP
System Model:              HP Laptop 15s-fq2xxx
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 140 Stepping 1 GenuineIntel ~2803 Mhz
BIOS Version:              AMI F.15, 9/15/2021
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC+01:00) Brussels, Copenhagen, Madrid, Paris
Total Physical Memory:     11,931 MB
Available Physical Memory: 7,140 MB
Virtual Memory: Max Size:  27,612 MB
Virtual Memory: Available: 18,264 MB
Virtual Memory: In Use:    9,348 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\LAPTOP-DE17RPQT
Hotfix(s):                 42 Hotfix(s) Installed.
                           [01]: KB5042097
                           [02]: KB5028951
                           [03]: KB4534170
                           [04]: KB4537759
                           [05]: KB4545706
                           [06]: KB4562830
                           [07]: KB4586864
                           [08]: KB5003791
                           [09]: KB5011048
                           [10]: KB5012170
                           [11]: KB5015684
                           [12]: KB5008575
                           [13]: KB5043064
                           [14]: KB5006753
                           [15]: KB5007273
                           [16]: KB5011352
                           [17]: KB5011651
                           [18]: KB5014035
                           [19]: KB5014671
                           [20]: KB5015895
                           [21]: KB5016705
                           [22]: KB5018506
                           [23]: KB5020372
                           [24]: KB5022924
                           [25]: KB5023794
                           [26]: KB5025315
                           [27]: KB5026879
                           [28]: KB5028318
                           [29]: KB5028380
                           [30]: KB5029709
                           [31]: KB5031539
                           [32]: KB5032392
                           [33]: KB5032907
                           [34]: KB5034224
                           [35]: KB5036447
                           [36]: KB5037018
                           [37]: KB5037240
                           [38]: KB5037995
                           [39]: KB5039336
                           [40]: KB5041579
                           [41]: KB5043935
                           [42]: KB5005699
Network Card(s):           2 NIC(s) Installed.
                           [01]: Intel(R) Wi-Fi 6 AX201 160MHz
                                 Connection Name: Wi-Fi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.1.1
                                 IP address(es)
                                 [01]: 192.168.1.116
                                 [02]: fe80::d585:3ad8:bb56:d6b4
                           [02]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
Hyper-V Requirements:      VM Monitor Mode Extensions: Yes
                           Virtualization Enabled In Firmware: Yes
                           Second Level Address Translation: Yes
                           Data Execution Prevention Available: Yes

### Build issue description

I have intel hardware so I wanted to try running a llm using OpenVINO following this youtube video example:
[Hugging Face + OpenVINO](https://www.youtube.com/watch?v=TsBBueu3AAI)

I've made a few changes to the code so I'll leave the full file bellow, but all the installed packages (used in the video) should be the latest as of today.

When the model loading completes the weight compression it throws and error: *PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\{user}\\AppData\\Local\\Temp\\tmpqs8oj2zf\\openvino_model.bin'*

Once the code errors out it clears all the temp files, so there is no way for me to check the "openvino_model.bin" file.
I've tried enabling developer mode, running python on admin mode and changing multiple model loading parameters like: "export, compile, etc...". I've also tried clearing all the saved models using "huggingface-cli delete-cache" and rebooting my PC, but even with no models saved I still get the error.

Any help or guidance is appreciated 😄 

### Build script or step-by-step to reproduce

```python
from optimum.intel import OVModelForCausalLM
from transformers import AutoTokenizer, pipeline

model_id = "mistralai/Mistral-7B-Instruct-v0.3"
model = OVModelForCausalLM.from_pretrained(model_id, export = True, compile = False)
model.save_pretrained("/ov_model");
tokenizer = AutoTokenizer.from_pretrained(model_id, clean_up_tokenization_spaces = True)
tokenizer.save_pretrained("/ov_model");
pipe = pipeline("text-generation-with-past", model = model, tokenizer = tokenizer)
outputs = pipe('User: Hello, how are you?\nAssitant:')

print(outputs)
```

This is my exact python file.

### Relevant log output

```shell
Traceback (most recent call last):
  File "D:\Dev\python_intel_llms\app.py", line 6, in <module>
    model = OVModelForCausalLM.from_pretrained(model_id, export = True, compile = False)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Dev\python_intel_llms\_env\Lib\site-packages\optimum\intel\openvino\modeling_base.py", line 373, in from_pretrained
    return super().from_pretrained(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Dev\python_intel_llms\_env\Lib\site-packages\optimum\modeling_base.py", line 424, in from_pretrained
    return from_pretrained_method(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Dev\python_intel_llms\_env\Lib\site-packages\optimum\intel\openvino\modeling_decoder.py", line 285, in _from_transformers
    main_export(
  File "D:\Dev\python_intel_llms\_env\Lib\site-packages\optimum\exporters\openvino\__main__.py", line 432, in main_export
    submodel_path.with_suffix(".bin").unlink()
  File "D:\Langs\Python\Lib\pathlib.py", line 1342, in unlink
    os.unlink(self)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\{user}\\AppData\\Local\\Temp\\tmpqs8oj2zf\\openvino_model.bin'
Traceback (most recent call last):
  File "D:\Langs\Python\Lib\shutil.py", line 633, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\{user}\\AppData\\Local\\Temp\\tmpqs8oj2zf\\openvino_model.bin'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\Langs\Python\Lib\weakref.py", line 666, in _exitfunc
    f()
  File "D:\Langs\Python\Lib\weakref.py", line 590, in __call__
    return info.func(*info.args, **(info.kwargs or {}))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Langs\Python\Lib\tempfile.py", line 935, in _cleanup
    cls._rmtree(name, ignore_errors=ignore_errors)
  File "D:\Langs\Python\Lib\tempfile.py", line 930, in _rmtree
    _shutil.rmtree(name, onexc=onexc)
  File "D:\Langs\Python\Lib\shutil.py", line 781, in rmtree
    return _rmtree_unsafe(path, onexc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Langs\Python\Lib\shutil.py", line 635, in _rmtree_unsafe
    onexc(os.unlink, fullname, err)
  File "D:\Langs\Python\Lib\tempfile.py", line 905, in onexc
    _os.unlink(path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\{user}\\AppData\\Local\\Temp\\tmpqs8oj2zf\\openvino_model.bin'
```

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.

**Comment by AlexKoff88** [comment]
Hi @iker-munoz-molero, this is a known problem on Windows. It was fixed some time ago: https://github.com/python/cpython/pull/113128/files

The solution is to update your Python to the latest version. For example, I don't see the issue on my Windows 11 machine with Python 3.12.6. Can you please try?

**Comment by richard087** [comment]
> The solution is to update your Python to the latest version. For example, I don't see the issue on my Windows 11 machine with Python 3.12.6. Can you please try?


I've replicated the problem. I've used a new install of Python 3.12 from the MS Store, on win11 (https://www.microsoft.com/store/productId/9NCVDN91XZQP?ocid=pdpshare)

process:
```
python -m venv p3.12-ov
cd p3.12-ov
.\Scripts\Activate.ps1
pip install optimum-intel openvino psutil nncf
```
execute the script given above

result:
```
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\tempfile.py", line 905, in onexc
    _os.unlink(path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\USER\\AppData\\Local\\Temp\\tmppiweeqre\\openvino_model.bin'
```

**Comment by richard087** [comment]
> Hi @iker-munoz-molero, this is a known problem on Windows. It was fixed some time ago: https://github.com/python/cpython/pull/113128/files
> 
This appears to have made it into 3.11 as well - https://github.com/python/cpython/commit/220fae770642141d91ca6299afd775493979c54b

**Comment by Nicogs43** [comment]
Yesterday I encountered the same error. I am using python 3.11.9 in my virtual env.

**Comment by nikita-savelyevv** [comment]
@iker-munoz-molero @richard087 @Nicogs43 Could you please try it with OpenVINO ver. 2024.3?
```
pip uninstall -y openvino openvino-tokenizers
pip install openvino==2024.3 openvino-tokenizers[transformers]==2024.3
```

**Comment by Nicogs43** [comment]
> @iker-munoz-molero @richard087 @Nicogs43 Could you please try it with OpenVINO ver. 2024.3?
> 
> ```
> pip uninstall -y openvino openvino-tokenizers
> pip install openvino==2024.3 openvino-tokenizers[transformers]==2024.3
> ```

I already use version 2024.3.0 of OpenVINO.

**Comment by richard087** [comment]
> @iker-munoz-molero @richard087 @Nicogs43 Could you please try it with OpenVINO ver. 2024.3?
> 
> ```
> pip uninstall -y openvino openvino-tokenizers
> pip install openvino==2024.3 openvino-tokenizers[transformers]==2024.3
> ```

At start:
```code\3.12-ov> pip show openvino
Name: openvino
Version: 2024.4.0
Summary: OpenVINO(TM) Runtime
Home-page: https://docs.openvino.ai/2023.0/index.html
Author: Intel(R) Corporation
Author-email: openvino@intel.com
License: OSI Approved :: Apache Software License
Location: c:\Users\richardcottrill\code\3.12-ov\.venv\Lib\site-packages
Requires: numpy, openvino-telemetry, packaging
Required-by: 
```

```
> pip uninstall -y openvino openvino-tokenizers
Found existing installation: openvino 2024.4.0
Uninstalling openvino-2024.4.0:
  Successfully uninstalled openvino-2024.4.0
WARNING: Skipping openvino-tokenizers as it is not installed.
> pip install openvino==2024.3 openvino-tokenizers[transformers]==2024.3
...
Successfully installed openvino-2024.3.0 openvino-tokenizers-2024.3.0.0 tiktoken-0.8.0
```

On running the same test script, I see the same result:
```
...
  File "C:\Users\USER\code\3.12-ov\.venv\Lib\site-packages\optimum\exporters\openvino\__main__.py", line 432, in main_export
    submodel_path.with_suffix(".bin").unlink()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\pathlib.py", line 1342, in unlink
    os.unlink(self)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\USER\\AppData\\Local\\Temp\\tmplih8nwea\\openvino_model.bin'
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\shutil.py", line 633, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\USER\\AppData\\Local\\Temp\\tmplih8nwea\\openvino_model.bin'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\weakref.py", line 666, in _exitfunc
    f()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\weakref.py", line 590, in __call__
    return info.func(*info.args, **(info.kwargs or {}))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\tempfile.py", line 935, in _cleanup
    cls._rmtree(name, ignore_errors=ignore_errors)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\tempfile.py", line 930, in _rmtree
    _shutil.rmtree(name, onexc=onexc)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\shutil.py", line 781, in rmtree
    return _rmtree_unsafe(path, onexc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\shutil.py", line 635, in _rmtree_unsafe
    onexc(os.unlink, fullname, err)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\tempfile.py", line 905, in onexc
    _os.unlink(path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\USER\\AppData\\Local\\Temp\\tmplih8nwea\\openvino_model.bin'
```

**Comment by richard087** [comment]
> ```
> python -m pip install git+https://github.com/huggingface/optimum-intel.git
> ```
Tried this in the environment from https://github.com/openvinotoolkit/openvino/issues/26844#issuecomment-2387725291 . the command resulted in:
```
...
Building wheels for collected packages: optimum-intel
  Building wheel for optimum-intel (pyproject.toml) ... done
  Created wheel for optimum-intel: filename=optimum_intel-1.21.0.dev0+5f539d5-py3-none-any.whl size=262727 sha256=55c9341d06f105f3dd731eb4cf5f617c75af2e2460585bb967437cc094289024
  Stored in directory: C:\Users\richardcottrill\AppData\Local\Temp\pip-ephem-wheel-cache-yq_0tm84\wheels\3e\28\37\922d7ce51fa59781a14b4187a13184af5a1d19da8b80d59adb
Successfully built optimum-intel
Installing collected packages: optimum, optimum-intel
  Attempting uninstall: optimum
    Found existing installation: optimum 1.22.0
    Uninstalling optimum-1.22.0:
      Successfully uninstalled optimum-1.22.0
  Attempting uninstall: optimum-intel
    Found existing installation: optimum-intel 1.19.0
    Uninstalling optimum-intel-1.19.0:
      Successfully uninstalled optimum-intel-1.19.0
Successfully installed optimum-1.23.3 optimum-intel-1.21.0.dev0+5f539d5
```
Re-ran the test (as above). Same result.

**Comment by riatzukiza** [comment]
I ran into this problem as well and have since attempted all the suggested steps above and still have not resolved it. I am using python 3.12.7, and using pipenv to manage my virtual environments.

**Comment by mikewastedmoney** [comment]
Anybody ever have luck resolving this issue?

**Comment by nikita-savelyevv** [comment]
@richard087 @Nicogs43 @riatzukiza @mikewastedmoney, please try again on the latest optimum-intel main branch version. We've recently merged another PR that should addresses this issue https://github.com/huggingface/optimum-intel/pull/1028 .

**Comment by richard087** [comment]
A different behavior (still a failure). Using the same process as https://github.com/openvinotoolkit/openvino/issues/26844#issuecomment-2445676946 
Output below:

> PS C:\Users\username\code\3.12-ov> python -m pip install git+https://github.com/huggingface/optimum-intel.git
> Collecting git+https://github.com/huggingface/optimum-intel.git
>   Cloning https://github.com/huggingface/optimum-intel.git to c:\users\username\appdata\local\temp\pip-req-build-9tgbzjel
>   Running command git clone --filter=blob:none --quiet https://github.com/huggingface/optimum-intel.git 'C:\Users\username\AppData\Local\Temp\pip-req-build-9tgbzjel'
>   Resolved https://github.com/huggingface/optimum-intel.git to commit c454b0000279ac9801302d726fbbbc1152733315
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... done
>   Preparing metadata (pyproject.toml) ... done
> Requirement already satisfied: torch>=1.11 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel==1.21.0.dev0+c454b00) (2.4.1)
> Requirement already satisfied: optimum~=1.23 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel==1.21.0.dev0+c454b00) (1.23.3)
> Requirement already satisfied: transformers<4.47,>=4.36 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel==1.21.0.dev0+c454b00) (4.44.2)
> Requirement already satisfied: datasets>=1.4.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel==1.21.0.dev0+c454b00) (3.0.1)
> Requirement already satisfied: sentencepiece in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel==1.21.0.dev0+c454b00) (0.2.0)
> Requirement already satisfied: setuptools in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel==1.21.0.dev0+c454b00) (75.1.0)
> Requirement already satisfied: scipy in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel==1.21.0.dev0+c454b00) (1.14.1)
> Requirement already satisfied: onnx in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel==1.21.0.dev0+c454b00) (1.17.0)
> Requirement already satisfied: filelock in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (3.16.1)
> Requirement already satisfied: numpy>=1.17 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (1.26.4)
> Requirement already satisfied: pyarrow>=15.0.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (17.0.0)
> Requirement already satisfied: dill<0.3.9,>=0.3.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (0.3.8)
> Requirement already satisfied: pandas in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2.2.3)
> Requirement already satisfied: requests>=2.32.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2.32.3)
> Requirement already satisfied: tqdm>=4.66.3 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (4.66.5)
> Requirement already satisfied: xxhash in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (3.5.0)
> Requirement already satisfied: multiprocess in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (0.70.16)
> Requirement already satisfied: fsspec<=2024.6.1,>=2023.1.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from fsspec[http]<=2024.6.1,>=2023.1.0->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2024.6.1)
> Requirement already satisfied: aiohttp in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (3.10.8)
> Requirement already satisfied: huggingface-hub>=0.22.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (0.25.1)
> Requirement already satisfied: packaging in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (24.1)
> Requirement already satisfied: pyyaml>=5.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (6.0.2)
> Requirement already satisfied: coloredlogs in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum~=1.23->optimum-intel==1.21.0.dev0+c454b00) (15.0.1)
> Requirement already satisfied: sympy in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum~=1.23->optimum-intel==1.21.0.dev0+c454b00) (1.13.3)
> Requirement already satisfied: typing-extensions>=4.8.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from torch>=1.11->optimum-intel==1.21.0.dev0+c454b00) (4.12.2)
> Requirement already satisfied: networkx in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from torch>=1.11->optimum-intel==1.21.0.dev0+c454b00) (3.3)
> Requirement already satisfied: jinja2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from torch>=1.11->optimum-intel==1.21.0.dev0+c454b00) (3.1.4)
> Requirement already satisfied: regex!=2019.12.17 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from transformers<4.47,>=4.36->optimum-intel==1.21.0.dev0+c454b00) (2024.9.11)
> Requirement already satisfied: safetensors>=0.4.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from transformers<4.47,>=4.36->optimum-intel==1.21.0.dev0+c454b00) (0.4.5)
> Requirement already satisfied: tokenizers<0.20,>=0.19 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from transformers<4.47,>=4.36->optimum-intel==1.21.0.dev0+c454b00) (0.19.1)
> Requirement already satisfied: protobuf>=3.20.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from onnx->optimum-intel==1.21.0.dev0+c454b00) (5.28.2)
> Requirement already satisfied: aiohappyeyeballs>=2.3.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2.4.3)
> Requirement already satisfied: aiosignal>=1.1.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (1.3.1)
> Requirement already satisfied: attrs>=17.3.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (24.2.0)
> Requirement already satisfied: frozenlist>=1.1.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (1.4.1)
> Requirement already satisfied: multidict<7.0,>=4.5 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (6.1.0)
> Requirement already satisfied: yarl<2.0,>=1.12.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (1.13.1)
> Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from requests>=2.32.2->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (3.3.2)
> Requirement already satisfied: idna<4,>=2.5 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from requests>=2.32.2->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (3.10)
> Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from requests>=2.32.2->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2.2.3)
> Requirement already satisfied: certifi>=2017.4.17 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from requests>=2.32.2->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2024.8.30)
> Requirement already satisfied: colorama in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from tqdm>=4.66.3->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (0.4.6)
> Requirement already satisfied: humanfriendly>=9.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from coloredlogs->optimum~=1.23->optimum-intel==1.21.0.dev0+c454b00) (10.0)
> Requirement already satisfied: MarkupSafe>=2.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from jinja2->torch>=1.11->optimum-intel==1.21.0.dev0+c454b00) (2.1.5)
> Requirement already satisfied: python-dateutil>=2.8.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pandas->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2.9.0.post0)
> Requirement already satisfied: pytz>=2020.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pandas->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2024.2)
> Requirement already satisfied: tzdata>=2022.7 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pandas->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (2024.2)
> Requirement already satisfied: mpmath<1.4,>=1.1.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from sympy->optimRequirement already satisfied: six>=1.5 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from python-dateutil>=2.8.2->pandas->datasets>=1.4.0->optimum-intel==1.21.0.dev0+c454b00) (1.16.0)
> Building wheels for collected packages: optimum-intel
>   Building wheel for optimum-intel (pyproject.toml) ... done
>   Created wheel for optimum-intel: filename=optimum_intel-1.21.0.dev0+c454b00-py3-none-any.whl size=279779 sha256=a0e15a11fcdd68430237d78505f451d15ebcf2ac65e363fcb0f8e6b8384779a8
>   Stored in directory: C:\Users\username\AppData\Local\Temp\pip-ephem-wheel-cache-306z7m8u\wheels\3e\28\37\922d7ce51fa59781a14b4187a13184af5a1d19da8b80d59adb
> Successfully built optimum-intel
> Installing collected packages: optimum-intel
>   Attempting uninstall: optimum-intel
>     Found existing installation: optimum-intel 1.21.0.dev0+5f539d5
>     Uninstalling optimum-intel-1.21.0.dev0+5f539d5:
>       Successfully uninstalled optimum-intel-1.21.0.dev0+5f539d5
> Successfully installed optimum-intel-1.21.0.dev0+c454b00
> PS C:\Users\username\code\3.12-ov> & c:/Users/username/code/3.12-ov/.venv/Scripts/python.exe c:/Users/username/code/3.12-ov/test.py
> Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00,  3.11it/s]
> The class `optimum.bettertransformers.transformation.BetterTransformer` is deprecated and will be removed in a future release.
> WARNING:root:Cannot apply model.to_bettertransformer because of the exception:
> The model type phi3 is not yet supported to be used with BetterTransformer. Feel free to open an issue at https://github.com/huggingface/optimum/issues if you would like this model type to be supported. Currently supported models are: dict_keys(['albert', 'bark', 'bart', 'bert', 'bert-generation', 'blenderbot', 'bloom', 'camembert', 'blip-2', 'clip', 'codegen', 'data2vec-text', 'deit', 'distilbert', 'electra', 'ernie', 'fsmt', 'gpt2', 'gptj', 'gpt_neo', 'gpt_neox', 'hubert', 'layoutlm', 'm2m_100', 'marian', 'markuplm', 'mbart', 'opt', 'pegasus', 'rembert', 'prophetnet', 'roberta', 'roc_bert', 'roformer', 'splinter', 'tapas', 't5', 'vilt', 'vit', 'vit_mae', 'vit_msn', 'wav2vec2', 'xlm-roberta', 'yolos']).. Usage model with stateful=True may be non-effective if model does not contain torch.functional.scaled_dot_product_attention
> C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\transformers\modeling_attn_mask_utils.py:114: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
>   if (input_shape[-1] > 1 or self.sliding_window is not None) and self.is_causal:
> C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\optimum\exporters\onnx\model_patcher.py:306: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
>   if past_key_values_length > 0:
> C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\transformers\models\phi3\modeling_phi3.py:260: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
>   if seq_len > self.original_max_position_embeddings:
> C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\nncf\torch\dynamic_graph\wrappers.py:86: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
>   op1 = operator(*args, **kwargs)
> Traceback (most recent call last):
>   File "c:\Users\username\code\3.12-ov\test.py", line 5, in <module>
>     model = OVModelForCausalLM.from_pretrained(model_id, export = True, compile = False)
>             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\optimum\intel\openvino\modeling_base.py", line 470, in from_pretrained
>     return super().from_pretrained(
>            ^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\optimum\modeling_base.py", line 438, in from_pretrained        
>     return from_pretrained_method(
>            ^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\optimum\intel\openvino\modeling_decoder.py", line 313, in _from_transformers
>     main_export(
>   File "C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\optimum\exporters\openvino\__main__.py", line 458, in main_export
>     _weight_only_quantization(submodel, quantization_config)
>   File "C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\optimum\intel\openvino\quantization.py", line 938, in _weight_only_quantization
>     return nncf.compress_weights(
>            ^^^^^^^^^^^^^^^^^^^^^^
> TypeError: compress_weights() got an unexpected keyword argument 'backup_mode'
> PS C:\Users\username\code\3.12-ov>

**Comment by richard087** [comment]
> @richard087 Please update your virtual environment with the following command `python -m pip install git+https://github.com/huggingface/optimum-intel.git#egg=optimum-intel[openvino]`

Again. different behaviour, still not working.

> PS C:\Users\username\code\3.12-ov> python -m pip install 'git+https://github.com/huggingface/optimum-intel.git#egg=optimum-intel[openvino]'
> DEPRECATION: git+https://github.com/huggingface/optimum-intel.git#egg=optimum-intel[openvino] contains an egg fragment with a non-PEP 508 name pip 25.0 will enforce this behaviour change. A possible replacement is to use the req @ url syntax, and remove the egg fragment. Discussion can be found at https://github.com/pypa/pip/issues/11617
> Collecting optimum-intel (from optimum-intel[openvino])
>   Cloning https://github.com/huggingface/optimum-intel.git to c:\users\username\appdata\local\temp\pip-install-m5mzu0nt\optimum-intel_65fcc26c63a14238b97604812b44e360
>   Running command git clone --filter=blob:none --quiet https://github.com/huggingface/optimum-intel.git 'C:\Users\username\AppData\Local\Temp\pip-install-m5mzu0nt\optimum-intel_65fcc26c63a14238b97604812b44e360'
>   Resolved https://github.com/huggingface/optimum-intel.git to commit c454b0000279ac9801302d726fbbbc1152733315
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... done
>   Preparing metadata (pyproject.toml) ... done
> Requirement already satisfied: torch>=1.11 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (2.4.1)
> Requirement already satisfied: optimum~=1.23 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (1.23.3)
> Requirement already satisfied: transformers<4.47,>=4.36 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (4.44.2)
> Requirement already satisfied: datasets>=1.4.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (3.0.1)
> Requirement already satisfied: sentencepiece in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (0.2.0)
> Requirement already satisfied: setuptools in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (75.1.0)
> Requirement already satisfied: scipy in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (1.14.1)
> Requirement already satisfied: onnx in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (1.17.0)
> Requirement already satisfied: nncf>=2.14.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (2.14.0)
> Requirement already satisfied: openvino>=2024.5.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (2024.5.0)
> Requirement already satisfied: openvino-tokenizers>=2024.5.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum-intel->optimum-intel[openvino]) (2024.5.0.0)
> Requirement already satisfied: filelock in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (3.16.1)
> Requirement already satisfied: numpy>=1.17 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (1.26.4)
> Requirement already satisfied: pyarrow>=15.0.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (17.0.0)
> Requirement already satisfied: dill<0.3.9,>=0.3.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (0.3.8)
> Requirement already satisfied: pandas in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2.2.3)
> Requirement already satisfied: requests>=2.32.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2.32.3)
> Requirement already satisfied: tqdm>=4.66.3 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (4.66.5)
> Requirement already satisfied: xxhash in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (3.5.0)
> Requirement already satisfied: multiprocess in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (0.70.16)
> Requirement already satisfied: fsspec<=2024.6.1,>=2023.1.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from fsspec[http]<=2024.6.1,>=2023.1.0->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2024.6.1)
> Requirement already satisfied: aiohttp in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (3.10.8)
> Requirement already satisfied: huggingface-hub>=0.22.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (0.25.1)
> Requirement already satisfied: packaging in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (24.1)
> Requirement already satisfied: pyyaml>=5.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (6.0.2)
> Requirement already satisfied: jsonschema>=3.2.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (4.23.0)
> Requirement already satisfied: jstyleson>=0.0.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (0.0.2)
> Requirement already satisfied: natsort>=7.1.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (8.4.0)
> Requirement already satisfied: networkx<=3.3,>=2.6 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (3.3)
> Requirement already satisfied: ninja<1.12,>=1.10.0.post2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (1.11.1.1)
> Requirement already satisfied: openvino-telemetry>=2023.2.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (2024.1.0)
> Requirement already satisfied: psutil in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (6.0.0)
> Requirement already satisfied: pydot<3.0.0,>=1.4.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (2.0.0)
> Requirement already satisfied: pymoo>=0.6.0.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (0.6.1.3)
> Requirement already satisfied: rich>=13.5.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (13.9.1)
> Requirement already satisfied: scikit-learn>=0.24.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (1.5.2)
> Requirement already satisfied: tabulate>=0.9.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (0.9.0)
> Requirement already satisfied: coloredlogs in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum~=1.23->optimum-intel->optimum-intel[openvino]) (15.0.1)
> Requirement already satisfied: sympy in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from optimum~=1.23->optimum-intel->optimum-intel[openvino]) (1.13.3)
> Requirement already satisfied: typing-extensions>=4.8.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from torch>=1.11->optimum-intel->optimum-intel[openvino]) (4.12.2)
> Requirement already satisfied: jinja2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from torch>=1.11->optimum-intel->optimum-intel[openvino]) (3.1.4)
> Requirement already satisfied: regex!=2019.12.17 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from transformers<4.47,>=4.36->optimum-intel->optimum-intel[openvino]) (2024.9.11)
> Requirement already satisfied: safetensors>=0.4.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from transformers<4.47,>=4.36->optimum-intel->optimum-intel[openvino]) (0.4.5)
> Requirement already satisfied: tokenizers<0.20,>=0.19 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from transformers<4.47,>=4.36->optimum-intel->optimum-intel[openvino]) (0.19.1)
> Requirement already satisfied: protobuf>=3.20.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from onnx->optimum-intel->optimum-intel[openvino]) (5.28.2)
> Requirement already satisfied: aiohappyeyeballs>=2.3.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2.4.3)
> Requirement already satisfied: aiosignal>=1.1.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (1.3.1)
> Requirement already satisfied: attrs>=17.3.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (24.2.0)
> Requirement already satisfied: frozenlist>=1.1.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (1.4.1)
> Requirement already satisfied: multidict<7.0,>=4.5 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (6.1.0)
> Requirement already satisfied: yarl<2.0,>=1.12.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from aiohttp->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (1.13.1)
> Requirement already satisfied: jsonschema-specifications>=2023.03.6 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from jsonschema>=3.2.0->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (2023.12.1)
> Requirement already satisfied: referencing>=0.28.4 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from jsonschema>=3.2.0->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (0.35.1)
> Requirement already satisfied: rpds-py>=0.7.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from jsonschema>=3.2.0->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (0.20.0)
> Requirement already satisfied: python-dateutil>=2.8.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pandas->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2.9.0.post0)
> Requirement already satisfied: pytz>=2020.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pandas->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2024.2)
> Requirement already satisfied: tzdata>=2022.7 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pandas->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2024.2)
> Requirement already satisfied: pyparsing>=3 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pydot<3.0.0,>=1.4.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (3.1.4)
> Requirement already satisfied: matplotlib>=3 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (3.9.2)
> Requirement already satisfied: autograd>=1.4 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (1.7.0)
> Requirement already satisfied: cma==3.2.2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (3.2.2)
> Requirement already satisfied: alive-progress in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (3.1.5)
> Requirement already satisfied: Deprecated in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (1.2.14)
> Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from requests>=2.32.2->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (3.3.2)
> Requirement already satisfied: idna<4,>=2.5 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from requests>=2.32.2->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (3.10)
> Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from requests>=2.32.2->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2.2.3)
> Requirement already satisfied: certifi>=2017.4.17 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from requests>=2.32.2->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (2024.8.30)
> Requirement already satisfied: markdown-it-py>=2.2.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from rich>=13.5.2->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (3.0.0)
> Requirement already satisfied: pygments<3.0.0,>=2.13.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from rich>=13.5.2->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (2.18.0)
> Requirement already satisfied: joblib>=1.2.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from scikit-learn>=0.24.0->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (1.4.2)
> Requirement already satisfied: threadpoolctl>=3.1.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from scikit-learn>=0.24.0->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (3.5.0)
> Requirement already satisfied: colorama in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from tqdm>=4.66.3->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (0.4.6)
> Requirement already satisfied: humanfriendly>=9.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from coloredlogs->optimum~=1.23->optimum-intel->optimum-intel[openvino]) (10.0)
> Requirement already satisfied: MarkupSafe>=2.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from jinja2->torch>=1.11->optimum-intel->optimum-intel[openvino]) (2.1.5)
> Requirement already satisfied: mpmath<1.4,>=1.1.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from sympy->optimum~=1.23->optimum-intel->optimum-intel[openvino]) (1.3.0)
> Requirement already satisfied: pyreadline3 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from humanfriendly>=9.1->coloredlogs->optimum~=1.23->optimum-intel->optimum-intel[openvino]) (3.5.4)
> Requirement already satisfied: mdurl~=0.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from markdown-it-py>=2.2.0->rich>=13.5.2->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (0.1.2)
> Requirement already satisfied: contourpy>=1.0.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from matplotlib>=3->pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (1.3.0)
> Requirement already satisfied: cycler>=0.10 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from matplotlib>=3->pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (0.12.1)
> Requirement already satisfied: fonttools>=4.22.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from matplotlib>=3->pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (4.54.1)
> Requirement already satisfied: kiwisolver>=1.3.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from matplotlib>=3->pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (1.4.7)
> Requirement already satisfied: pillow>=8 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from matplotlib>=3->pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (10.4.0)
> Requirement already satisfied: six>=1.5 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from python-dateutil>=2.8.2->pandas->datasets>=1.4.0->optimum-intel->optimum-intel[openvino]) (1.16.0)
> Requirement already satisfied: about-time==4.2.1 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from alive-progress->pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (4.2.1)
> Requirement already satisfied: grapheme==0.6.0 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from alive-progress->pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (0.6.0)
> Requirement already satisfied: wrapt<2,>=1.10 in c:\users\username\code\3.12-ov\.venv\lib\site-packages (from Deprecated->pymoo>=0.6.0.1->nncf>=2.14.0->optimum-intel->optimum-intel[openvino]) (1.16.0)
> 
> [notice] A new release of pip is available: 24.2 -> 24.3.1
> [notice] To update, run: python.exe -m pip install --upgrade pip
> PS C:\Users\username\code\3.12-ov> & c:/Users/username/code/3.12-ov/.venv/Scripts/python.exe c:/Users/username/code/3.12-ov/test.py
> Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00,  8.26it/s]
> The class `optimum.bettertransformers.transformation.BetterTransformer` is deprecated and will be removed in a future release.
> WARNING:root:Cannot apply model.to_bettertransformer because of the exception:
> The model type phi3 is not yet supported to be used with BetterTransformer. Feel free to open an issue at https://github.com/huggingface/optimum/issues if you would like this model type to be supported. Currently supported models are: dict_keys(['albert', 'bark', 'bart', 'bert', 'bert-generation', 'blenderbot', 'bloom', 'camembert', 'blip-2', 'clip', 'codegen', 'data2vec-text', 'deit', 'distilbert', 'electra', 'ernie', 'fsmt', 'gpt2', 'gptj', 'gpt_neo', 'gpt_neox', 'hubert', 'layoutlm', 'm2m_100', 'marian', 'markuplm', 'mbart', 'opt', 'pegasus', 'rembert', 'prophetnet', 'roberta', 'roc_bert', 'roformer', 'splinter', 'tapas', 't5', 'vilt', 'vit', 'vit_mae', 'vit_msn', 'wav2vec2', 'xlm-roberta', 'yolos']).. Usage model with stateful=True may be non-effective if model does not contain torch.functional.scaled_dot_product_attention
> C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\transformers\modeling_attn_mask_utils.py:114: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
>   if (input_shape[-1] > 1 or self.sliding_window is not None) and self.is_causal:
> C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\optimum\exporters\onnx\model_patcher.py:306: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
>   if past_key_values_length > 0:
> C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\transformers\models\phi3\modeling_phi3.py:260: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
>   if seq_len > self.original_max_position_embeddings:
> C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\nncf\torch\dynamic_graph\wrappers.py:85: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
>   op1 = operator(*args, **kwargs)
> INFO:nncf:Statistics of the bitwidth distribution:
> +---------------------------+-----------------------------+----------------------------------------+
> | Weight compression mode   | % all parameters (layers)   | % ratio-defining parameters (layers)   |
> +===========================+=============================+========================================+
> | int8_asym                 | 100% (130 / 130)            | 100% (130 / 130)                       |
> +---------------------------+-----------------------------+----------------------------------------+
> Applying Weight Compression ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% • 0:00:47 • 0:00:00
> Traceback (most recent call last):
>   File "c:\Users\username\code\3.12-ov\test.py", line 9, in <module>
>     pipe = pipeline("text-generation-with-past", model = model, tokenizer = tokenizer)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\transformers\pipelines\__init__.py", line 849, in pipeline     
>     normalized_task, targeted_task, task_options = check_task(task)
>                                                    ^^^^^^^^^^^^^^^^
>   File "C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\transformers\pipelines\__init__.py", line 533, in check_task   
>     return PIPELINE_REGISTRY.check_task(task)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\username\code\3.12-ov\.venv\Lib\site-packages\transformers\pipelines\base.py", line 1339, in check_task      
>     raise KeyError(
> KeyError: "Unknown task text-generation-with-past, available tasks are ['audio-classification', 'automatic-speech-recognition', 'depth-estimation', 'document-question-answering', 'feature-extraction', 'fill-mask', 'image-classification', 'image-feature-extraction', 'image-segmentation', 'image-to-image', 'image-to-text', 'mask-generation', 'ner', 'object-detection', 'question-answering', 'sentiment-analysis', 'summarization', 'table-question-answering', 'text-classification', 'text-generation', 'text-to-audio', 'text-to-speech', 'text2text-generation', 'token-classification', 'translation', 'video-classification', 'visual-question-answering', 'vqa', 'zero-shot-audio-classification', 'zero-shot-classification', 'zero-shot-image-classification', 'zero-shot-object-detection', 'translation_XX_to_YY']"

**Comment by nikita-savelyevv** [comment]
@richard087 It's hard to tell for sure what's the reason for this error without looking at the code.

**Comment by nikita-savelyevv** [comment]
Closing the issue as it seems to be resolved
